src/java/org/apache/cassandra/net/CustomParamsSerializer.java

```diff
@@ -0,0 +1,72 @@
+
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.net;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.db.TypeSizes;
+import org.apache.cassandra.io.IVersionedSerializer;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+
+class CustomParamsSerializer implements IVersionedSerializer<Map<String,ByteBuffer>>
+{
+    public static final CustomParamsSerializer serializer = new CustomParamsSerializer();
+
+    @Override
+    public void serialize(Map<String, ByteBuffer> t, DataOutputPlus out, int version) throws IOException
+    {
+        out.writeInt(t.size());
+        for (Map.Entry<String, ByteBuffer> e : t.entrySet())
+        {
+            out.writeUTF(e.getKey());
+            ByteBufferUtil.writeWithLength(e.getValue(), out);
+        }
+    }
+
+    @Override
+    public Map<String, ByteBuffer> deserialize(DataInputPlus in, int version) throws IOException
+    {
+        int entries = in.readInt();
+        Map<String, ByteBuffer> customParams = new HashMap<>(entries);
```

Comment:
You want to use `Maps.newHashMapWithExpectedSize()` here, to take load factor into account.

Reply:
yup, good catch.